scripts/segments_logging.py

```diff
@@ -0,0 +1,27 @@
+import pandas as pd
+import time
+from whylogs.core.datasetprofile import dataframe_profile
+from whylogs import get_or_create_session
+
+if __name__ == "__main__":
+    df = pd.read_csv("data/lending-club-accepted-10.csv")
+    print(df.head())
+    session = get_or_create_session()
+
+    with session.logger(
+        "segment", segments=[[{"key": "home_ownership", "value": "RENT"}], [{"key": "home_ownership", "value": "MORTGAGE"}]], cache=1
+    ) as logger:
+        print(session.get_config())
+        profile = logger.log_dataframe(df)
+
+    with session.logger("my_rotated_seg", segments=["home_ownership"], with_rotation_time="s", cache=1) as logger:
+        print(session.get_config())
+        profile = logger.log_dataframe(df)
+        time.sleep(2)
+        profile = logger.log_dataframe(df)
+
+    with session.logger("my_rotated_seg_two_keys", segments=["home_ownership", "sub_grade"], with_rotation_time="s", cache=1) as logger:
+        print(session.get_config())
+        profile = logger.log_csv("data/lending-club-accepted-10.csv")
+        time.sleep(2)
+        profile = logger.log_dataframe(df)
```

Comment:
just from the API perspective, what does `logger.log...` returns now that we pass in a list of segment keys?

Reply:
the "log" api is bit all over the place, in session "log_*" returns a profile, while in logger, it simply does that, log the data into the current active datasetprofile. These lines are an artifact, since it doest return anything those are None profiles. It came from a previous version, and i will remove it as to not create a confusion

Reply:
FYI, there's a use case of accessing the underlying dataset profiles when running whylogs without relying on the `writer` to write out the data. That's why the API returns them in the session (just makes it a little bit easier).

There might be an argument to have a `DatasetRun` or `DatasetProfilesBatch` object that contains all the segment data here, though the same is still a bit hazy to me

Reply:
You can acess all the current active profiles with `logger.profile` -> `DatasetProfile` or `logger.segmented_profiles`-> `Dict[str,DatasetProfile]`.